### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by illiashcherbak**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -1,61 +1,72 @@
...
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = getPeopleBySex(people, 'm');
+  const withCentury = century
```
Comment (OlggaJul) [inline src/averageAges.js:11]: but if century is not specified, this variable name will not be obvious

File: src/averageAges.js [hunk 2]
```diff
@@ -1,61 +1,72 @@
...
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = getPeopleBySex(people, 'f');
+  const hasChildren = withChildren
```
Comment (OlggaJul) [inline src/averageAges.js:40]: if they have no children this variable name will not be obvious

File: src/averageAges.js [hunk 3]
```diff
@@ -1,61 +1,72 @@
...
  *
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => (
+    people.find(person => person.name === child.mother)
+  ));
+
+  const hasOnlySon = onlyWithSon
```
Comment (OlggaJul) [inline src/averageAges.js:60]: the same)

**Review by OlggaJul — CHANGES_REQUESTED** [review]

**Comment by illiashcherbak** [comment]
Perhaps I didn't get your comment, so I just renamed the variables

**Review by OlggaJul — APPROVED** [review]

**Review by ivan-kadykalo — APPROVED** [review]
Good job 🔥

**Pull request closed without merging**